junit5/src/test/java/org/jboss/weld/junit5/auto/nested/OuterTestClassAndBeanSameInstanceTest.java

```diff
@@ -0,0 +1,81 @@
+package org.jboss.weld.junit5.auto.nested;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Objects;
+
+import javax.enterprise.event.Event;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Disposes;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import org.jboss.weld.junit5.auto.EnableAutoWeld;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests that outer test classes don't get instantiated again for {@code @}{@link Produces Produces},
+ * {@code @}{@link Disposes Disposes}, or {@code @}{@link Observes Observes} annotations.
+ * 
+ * Pay attention to with particular instance each outer class member variable reference would refer to
```

Comment:
This is really hard to read, please reformulate it a bit.

Reply:
You are right and I appreciate your comment very much. But then that is really difficult to express. I tried to express the core topic here of this MR in as few words as would not make the next guy reading it tired and English is not my native language. So that's a tough challenge and I try my best and also would not reject help either. I checked in a few minor changes reformulating it a tiny bit and then came up with an attempt to clarify with an example like this

> It would be wrong if Weld instantiated the outer test class again in order to invoke the producer method {@link #produceBean()} on it, for example.
> Another {@link OuterTestClassAndBeanSameInstanceTest} instance would get another {@link Bean} instance assigned to {@link #theBean}.
> The {@link OuterTestClassAndBeanSameInstanceTest} instance created by JUnit also has a producer method {@link #produceBean()} but would never be invoked by Weld,
> because Weld would use its own instance of the test class it would treat as a bean and instantiate not knowing JUnit already has instantiated it.
> In the test, member variables would have different values or point then to other instances than those Weld injects.
> All of this would result in confusion and this test here verifies that that does not happen.

but am not happy with it myself. Any suggestion?

Reply:
English is not my native lang either, but I can try and help. How about:
> Despite having several CDI constructs that need to reside inside a bean (producer, observer, ...), the outermost class is not supposed to be instantiated by Weld. Instead, it is only instantiated by JUnit and Weld then consumes and enriches that instance. This is verified by testing that the `Bean` references throughout the test are the same.

Reply:
This time I was having a hard time to understand your text. There are a few spongy terms and I guess I would have not understood it correctly without the pre-existing context. The structure makes good sense and I took it and kind of translated it into my own words and added more words and hope after this iteration it has improved. It also inspired a few more testcases all of which failed before.

I still don't agree to the "outermost" part but could not really avoid to mention it that way in my version and hope we can agree on a formula anyway. After all it's only a test comment but never the less I try to leave statements behind I think are true but I will not insist on this point forever.